tck/value-entity.js

```diff
@@ -55,7 +55,10 @@ function process(request, state, context) {
   effects.forEach(effect =>
     reply.addEffect(two.service.methods.Call, { id: effect.id }, effect.synchronous)
   )
-  return reply
+
+  return new Promise((resolve, reject) => {
+    setTimeout(() => resolve(reply), 10);
+  });
```

Comment:
I think it would be good if we could cover both async and non async in the TCK across all component callback types somehow. Or wdyt @andreaTP is the paths the same so it does not really matter? 

I'm thinking with only async we could easily end up break the logic allowing us to just return a reply directly in all these places without noticing (even if that works in this PR.


  I guess it can't be a single run because the TCK itself does not support that but, we have complete control over the runner `run.js` so that could either run both or take a flag or something?

Reply:
I changed just a couple of handlers to have integration tests for this feature; do you think I should revert?

There are still `commandHandlers` using the sync interface (that boils down to the same code-path now) e.g.:
https://github.com/lightbend/akkaserverless-javascript-sdk/blob/ab9f14f7adba43843710c7190e822ac6611d6a71/tck/value-entity.js#L72

Reply:
Ok, hmm, I'd sleep better with a complete coverage (it would also somewhat document where it is supported and where it is still not possible to return a promise). On the other hand that might be a lot of extra work, so maybe this is enough, not quite sure.

Reply:
I added a few integration tests to cover this change:
https://github.com/lightbend/akkaserverless-javascript-sdk/pull/43/commits/2471b51538339dcf72f06026314768d34e256a0b

Thanks for the push @johanandren , do you think that this is enough or we should cover more?